src/backend/api/controller.js

```diff
@@ -73,10 +73,110 @@ const deleteById = async (req, res, table) => {
   }
 };
 
+const getMealReview = async (req, res) => {
+  try {
+    const { meal_id } = req.params;
+    const data = await knex('review').where({ meal_id });
+    if (data.length > 0) {
+      res.status(200).json(data);
+    } else {
+      res.status(404).json({
+        'message': `review for meal with id ${req.params.meal_id} does not exist`
+      });
+    }
+  } catch (e) {
+    res.status(500).json(e.message);
+  }
+};
+
+const getMealByQuery = async (req, res) => {
+  try {
+    const queries = {};
+    const parameters = [
+      'maxPrice',
+      'availableReservations',
+      'title',
+      'dateAfter',
+      'dateBefore',
+      'limit',
+      'sortKey'
+    ];
+    // Put key value pairs where keys match parameters into queries object
+    Object.keys(req.query).forEach((key) => {
+      if (parameters.includes(key)) {
+        queries[key] = req.query[key];
+      }
+    });
+    if (Object.keys(queries).length === 0) {
+      getAllFromTable(req, res, 'meal');
+    } else {
+      // Build dynamic querie with all possible parameters
+      const meals = knex('meal')
```

Comment:
Here are a few ways to shorten and simplify this query:

1. Extract query conditions to helper functions:

```js
function filterByPrice(builder) {
  if (queries.maxPrice > 0) {
    builder.where('price', '<=', queries.maxPrice); 
  }
}

function filterByTitle(builder) {
  if (typeof queries.title === 'string') {
    builder.where('title', 'like', `%${queries.title}%`);
  } 
}

// etc for other filters
```

2. Use chaining instead of nested modifies:

```js
.modify(filterByPrice)
.modify(filterByTitle) 
.orderBy(getSort())
.having(getAvailabilityFilter())
.limit(getLimit())
```

3. Extract common query chunks to variables:

```js
const baseQuery = knex('meal')
  .select(...)
  .leftJoin(...)
  .groupBy(...)

baseQuery
  .modify(filterByPrice)
  .modify(filterByTitle)
  // etc
```

4. Use knex query builder helpers like .whereRaw() for complex bits:

```js 
.whereRaw('available_reservations > 0 OR available_reservations IS NULL')
```

5. Bind query values to avoid repetition:

```js
.where('title', 'like', knex.raw('?', ['%' + queries.title + '%']))
```

The key ideas are to break the query into smaller reusable pieces, use chaining over nested modifies, leverage query builder helpers, and bind values.